### PR TITLE
Rebar3 with fallback rebar2 support

### PR DESCRIPTION
**Pull request by jadeallenx**
This PR adds support for rebar3. Since this PR and my other OTP 19 PR #14 both edit rebar.config there will be a merge conflict but I'd be happy to rebase after you pick one to merge first. 😄

### CODE REVIEW

**Comment by jadeallenx** [comment]
Appears #5 is almost the same without fallback rebar2 support.

**Comment by jadeallenx** [comment]
Looks like rebar3 is supported by travis since https://github.com/travis-ci/travis-ci/issues/6506

**Comment by jadeallenx** [comment]
Probably also should update the README

**Comment by jadeallenx** [comment]
lol travis is failing saying rebar3 isn't on the path 😛

File: rebar.config.script [hunk 1]
```diff
@@ -0,0 +1,12 @@
+case erlang:function_exported(rebar3, main, 1) of
+    true -> % rebar3
+        CONFIG;
+    false -> % rebar 2.x or older
+        %% Rebuild deps, possibly including those that have been moved to
+        %% profiles
+        [{deps, [
+	    {bear, ".*", {git, "https://github.com/folsom-project/bear.git", {tag, "0.8.2"}}},
```
Comment (joewilliams) [inline rebar.config.script]: I tagged bear 0.8.3, please update this version.

**Comment by joewilliams** [comment]
Please fix the `rebar.config` conflict. Not sure why travis is complaining but otherwise I'll merge this and tag it.

**Comment by jadeallenx** [comment]
@joewilliams Thanks for the merge - found an issue that needs to be resolved though :) It has to do with meck and the API it calls for coverage in OTP 18 and 19

**Pull request merged**